### DISCUSSION TITLE
TimePickerWithHistory behaviour on different timezones

### DISCUSSION THREAD

**Discussion by lpskdl** [grafana/grafana#58737] [Dashboards, Panels & Visualizations]
There is an existing bug with our TimePickerWithHistory as it is not working correctly with different timezone are set and will always use the local browser's timezone. There is already an [escalation](https://github.com/grafana/support-escalations/issues/4432) related to this and [issue](https://github.com/grafana/grafana/issues/46806) which is also possible linked to this topic and possibly more will report it later.

We've seen that this component is used in 3 different places which are:
- `DashNavTimeControls`
- `ExploreTimeControls`
- `SceneTimePicker`

We would like to know and understand how the component is used in the following areas above especially:

- what tooltip will display in the timerange picker and panels when different timezone was set?
- when the user sets the timezone from the user's preference setting/dashboard setting. How will this affect the other areas of the app?
- how the dashboard panel will display the time based on these timezones levels(user preference, dashboard settings, etc.)
- how the absolute and relative time will behave when different timezones are set
other cases that we aren't aware of yet.

We will consolidate all the information we will get from this and plan how we can move forward with possible solution.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
